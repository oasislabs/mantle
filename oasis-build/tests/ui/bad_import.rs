#[derive(oasis_std::Service)]
pub struct Service;

impl Service {
    pub fn new(_ctx: &oasis_std::Context) -> Self {
        Self
    }

    pub fn bad_import(&mut self, _ctx: &oasis_std::Context, arg: xcc::NonXccType) {}
}

fn main() {
    oasis_std::service!(Service);
}
