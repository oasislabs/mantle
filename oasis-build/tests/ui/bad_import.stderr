error: External type `xcc::NonXccType` must be defined in a service to use in an RPC interface.
 --> $DIR/bad_import.rs:9:66
  |
9 |     pub fn bad_import(&mut self, _ctx: &oasis_std::Context, arg: xcc::NonXccType) {}
  |                                                                  ^^^^^^^^^^^^^^^

error: aborting due to previous error

